ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/coordinator/ArjunaTransactionImple.java

```diff
@@ -501,18 +491,6 @@ public void rollback () throws SystemException
 
 			status = super.status();
 		}
-		else
```

Comment:
But I agree that there is an issue in the rollback case, namely that trying to rollback an already aborted transaction should not throw an exception and instead should quietly succeed.

However, I am not convinced your code only addresses this requirement because the change introduces more changes than was called for:

* INVALID and CLEANUP now throw UNKNOWN whereas before we threw INVALID_TRANSACTION
* DISABLED, NO_ACTION and CREATED now return silently whereas before we threw INVALID_TRANSACTION

So I have the same comment, we err on the side of caution and usually reject behaviour change unless there are solid reasons to do so.

We can get a second opinion if required but my advice as reviewer is to only address the stated problem and no more and do something along the lines that I suggested earlier.